src/persist-client/src/batch.rs

```diff
@@ -381,24 +382,19 @@ where
                 upper: registered_upper,
             });
         }
-        // when since is less-equal to upper, the upper is a strict bound on the updates' timestamp.
-        // because user batches would never have a since in advance of upper, this ensures that new
+        // when since is equal to lower, the upper is a strict bound on the updates' timestamp.
+        // because user batches would never have a since in advance of lower, this ensures that new
         // updates are recorded with valid timestamps
-        if PartialOrder::less_than(&self.since, &registered_upper) {
-            if registered_upper.less_equal(&self.max_ts) {
-                return Err(InvalidUsage::UpdateBeyondUpper {
-                    max_ts: self.max_ts,
-                    expected_upper: registered_upper.clone(),
-                });
+        // Otherwise we can make no assumptions about the timestamps
+        if self.since == self.lower {
```

Comment:
to me the "common case" for user batches is that the since is `Antichain::from_elem(T::minimum())`, which I think this doesn't work for (hence the unit test failure). it feels kinda bad (because timely/dd don't do this sort thing much), but could this check be that _all elements of since_ be <= lower?

Reply:
Yes, that sounds like the correct check here. I changed the PR